### PR TITLE
test: add a custom command based log_upload workflow

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Add a system tests which can demonstrate how to handle command based log_upload commands by overriding the default log_upload workflow, and add a pre-processing step which will generate the log file when called (the command could do something like talk to a database etc.)

It shows a working example that can be referenced to customers.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/RobotFramework/tests/cumulocity/log/workflow/tedge-log-plugin.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[[files]]
+path = "/opt/homebrew/var/log/tedge/agent/software-*"
```
[the remark this answers is not in the document]
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/workflow/tedge-log-plugin.toml:2]: yeah sorry I overlooked that path.

[the remark this answers is not in the document]
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/workflow/tedge-log-plugin.toml]: Resolved by 4865f43714ad59e2058a24cae36c7300793673c6

File: tests/RobotFramework/tests/cumulocity/log/workflow/log_upload.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+on_error = "failed"
+
+[init]
+  action = "proceed"
+  on_success = "preprocess"
+
+[preprocess]
+  script = "/usr/bin/log_upload.sh ${.payload.status} ${.payload.type} ${.payload.tedgeUrl} ${.payload.dateFrom} ${.payload.dateTo} ${.payload.searchText}"
+  on_success = "scheduled"
+  on_error = "scheduled"
```
[the remark this answers is not in the document]
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/workflow/log_upload.toml:15]: Resolved by 4865f43714ad59e2058a24cae36c7300793673c6

I changed the message slightly to show the real intent (e.g. to keep the status transitions, as when using `on_error = "failed"`, the operation does status remains in `PENDING`)

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by gligorisaev — APPROVED** [review]
The Test is doing what is supposed to do

**Pull request merged**